src/app/digital/actions/ports/MuxPortChangeAction.ts

```diff
@@ -12,31 +12,62 @@ import {Multiplexer} from "digital/models/ioobjects/other/Multiplexer";
 import {InputPortChangeAction} from "./InputPortChangeAction";
 import {OutputPortChangeAction} from "./OutputPortChangeAction";
 
+
+/**
+ * This code changes the size of the Mux object based on how many inputs are entered. 
+ * When the selector inputs are increased the number of inputs are also increased by 2 to the power of the number of selector inputs chosen.
+ * Ex.) input count = 3, then the number of inputs changes to 2^3 or 8.
+ * The actual size of the mux object is also changed accordingly.
+ */
 export class MuxPortChangeAction extends PortChangeAction {
     protected obj: Mux;
 
     protected otherPortAction: PortChangeAction;
 
+    /**
+     * Either changes the size of the inputs if it's a multiplexor or the outputs if it's a Demux.
+     *
+     * @param obj refers to the Mux object
+     * @param initial refers to the initial number of inputs
+     * @param target refers to the new number of inputs requested
+     */
     public constructor(obj: Mux, initial: number, target: number) {
         super(obj.getDesigner(), target, initial);
         this.obj = obj;
 
         if (obj instanceof Multiplexer)
             this.otherPortAction = new InputPortChangeAction(obj, Math.pow(2, initial), Math.pow(2, target));
         else
-            this.otherPortAction = new OutputPortChangeAction(obj, Math.pow(2, initial), Math.pow(2, target));
+            this.otherPortAction = new OutputPortChangeAction(obj, Math.pow(2, initial), Math.pow(3, target));
```

Comment:
Is this line supposed to be changed?

Reply:
No I think I accidentally did that, I will change it back